src/data-entry.js

```diff
@@ -1,23 +1,27 @@
 // @flow
 import React, {Component, PropTypes as t} from 'react';
 import {handleError} from './error';
+import DataInput from './data-input';
 
 type EventType = {
   target: {
+    name: string,
     value: string
   }
 };
 
-async function addProject(name) {
-  const url = `${window.BASE_URL}/project?name=${name}`;
+async function addProject(props) {
+  const {name, description} = props;
+  let url = `${window.BASE_URL}/project?name=${name}`;
```

Comment:
Go ahead and create the function that Chris suggested. You can put that in a separate file, maybe named `url-util.js`. It could export a function named `getUrl` that takes a path string and query object. For example,
````
import {getUrl} from './url-util';
...
const url = getUrl('project', {name, description});
````
That function can prepend `window.BASE_URL` and append the query parameters.